iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/function_ref.inl

```diff
@@ -52,7 +53,7 @@ inline function_ref<ReturnType(ArgTypes...)>::function_ref(CallableType&& callab
 template <class ReturnType, class... ArgTypes>
 inline function_ref<ReturnType(ArgTypes...)>::function_ref(ReturnType (&function)(ArgTypes...)) noexcept
     // the cast is required to work on POSIX systems
-    // AXIVION Next Construct AutosarC++19_03-A5.2.4, AutosarC++19_03-A5.2.4-M5.2.6 : Type-safety ensured by casting
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4, AutosarC++19_03-M5.2.6 : Type-safety ensured by casting
```

Comment:
In my opinion this is not a justification. Here you have to argue that this is a template construct, the original type is restored via a cast to the original type with a callback which is set in the constructor itself, see line 67.

Furthermore you could mention that this is encapsulated in the `function_ref` to provide modern callback functionality to the user.

Basically, I would always state: why we want it and why it is safe in this case.